File: Leetcode/495. Teemo Attacking/solution.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        # get time posined for last number (always duration)
        timePoisoned = duration

        # get time posined for previous numbers
        for i in range(1, len(timeSeries)):
            diff = timeSeries[i] - timeSeries[i-1]
            timePoisoned += duration if diff > duration else diff

        return timePoisoned

File: Leetcode/495. Teemo Attacking/solution.rs
impl Solution {
    pub fn find_poisoned_duration(time_series: Vec<i32>, duration: i32) -> i32 {
        let mut time_poisoned = duration;
        
        for i in 1..time_series.len() {
            let diff = time_series[i] - time_series[i-1];
            time_poisoned += if diff > duration { duration } else { diff };
        }
        
        time_poisoned
    }
}